File: furniture-website-FE/api/CategoryApi.js
import { getApiUrl } from '../utils/Utils.js';
function CategoryApi(app) {
    app.factory('CategoryApi', function ($http) {
        return {
            filter(params) {
                return $http.get(getApiUrl('/public/categories/filter'), {
                    params
                });
            },
            getExistCategory() {
                return $http.get(getApiUrl('/public/categories-exist'));
            },
            createCategory(category) {
                return $http.post(getApiUrl('/admin/categories'), category, {
                    headers: {
                        Authorization: 'Bearer ' + localStorage.getItem('access_token')
                    }
                });
            },
            updateCategory(category) {
                return $http.put(getApiUrl('/admin/categories'), category, {
                    headers: {
                        Authorization: 'Bearer ' + localStorage.getItem('access_token')
                    }
                });
            },
            deleteCategory(id) {
                return $http.delete(getApiUrl('/admin/categories/' + id), {
                    headers: {
                        Authorization: 'Bearer ' + localStorage.getItem('access_token')
                    }
                });
            }
        };
    });
}

export default CategoryApi;

File: furniture-website-BE/src/main/java/com/poly/api/CategoryApi.java
package com.poly.api;

import com.poly.entity.Category;
import com.poly.payload.request.CategoryQueryParam;
import com.poly.service.CategoryService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api")
public class CategoryApi {
    @Autowired
    CategoryService categoryService;

    @GetMapping("public/categories/filter")
    public ResponseEntity<?> filterCategory(CategoryQueryParam categoryQueryParam) {
        return ResponseEntity.ok(categoryService.filterCategory(categoryQueryParam));
    }

    @GetMapping("/public/categories-exist")
    public ResponseEntity<?> getExistCategory() {
        return ResponseEntity.ok(categoryService.getExistCategory());
    }

    @PostMapping("/admin/categories")
    public ResponseEntity<?> createCategory(@RequestBody Category category) {
        return ResponseEntity.ok(categoryService.create(category));
    }

    @PutMapping("/admin/categories")
    public ResponseEntity<?> updateCategory(@RequestBody Category category) {
        return ResponseEntity.ok(categoryService.update(category));
    }

    @DeleteMapping("/admin/categories/{id}")
    public ResponseEntity<?> deleteCategory(@PathVariable(name = "id") Long id) {
        return ResponseEntity.ok(categoryService.delete(id));
    }
}


